### PR TITLE
[Merged by Bors] - feat: Fourier inversion formula

### PR DESCRIPTION
**Pull request by sgouezel**
We show the Fourier inversion formula on finite-dimensional real inner product spaces: if `f` and its Fourier transform are both integrable, then `𝓕⁻ (𝓕 f) = f` at continuity points of `f`.

---

- [x] depends on: #10829 
- [x] depends on: #10833 
- [x] depends on: #11035 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Fourier/Inversion.lean [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      congr 3
+      simp only [ofReal_mul, ofReal_ofNat]
+      ring
+    · simp [fourierIntegralInv_eq]
+  have B : Tendsto (fun (c : ℝ) ↦ (∫ w : V,
+        𝓕 (fun w ↦ cexp (- c⁻¹ * ‖w‖^2 + 2 * π * Complex.I * ⟪v, w⟫)) w • f w)) atTop
+      (𝓝 (𝓕⁻ (𝓕 f) v)) := by
+    apply A.congr'
+    filter_upwards [Ioi_mem_atTop 0] with c (hc : 0 < c)
+    have I : Integrable (fun w ↦ cexp (- c⁻¹ * ‖w‖^2 + 2 * π * Complex.I * ⟪v, w⟫)) :=
```
Comment (jcommelin) [inline Mathlib/Analysis/Fourier/Inversion.lean:91]: If `I` will just refer to the complex number, then `have I` should be updated a bit.

File: Mathlib/Analysis/Fourier/Inversion.lean [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    simp [div_eq_inv_mul]
+  · norm_cast
+    simp only [one_div, norm_smul, Real.norm_eq_abs, mul_pow, _root_.sq_abs, neg_mul, neg_inj,
+      ← rpow_nat_cast, ← rpow_mul hc.le, mul_assoc]
+    norm_num
+
+/-- **Fourier inversion formula**: If a function `f` on a finite-dimensional real inner product
+space is integrable, and its Fourier transform `𝓕 f` is also integrable, then `𝓕⁻ (𝓕 f) = f` at
+continuity points of `f`. -/
+theorem fourier_inversion (hf : Integrable f) (h'f : Integrable (𝓕 f)) {v : V}
```
Comment (jcommelin) [inline Mathlib/Analysis/Fourier/Inversion.lean:154]: Does it make sense to state a version for continuous `f` that gives equality of functions? (I.e., after `funext`, apply this current version.)

File: docs/undergrad.yaml [hunk 3]
```diff
@@ -532,8 +532,10 @@ Measures and integral calculus:
     Dirichlet theorem: ''
     Fejer theorem: ''
     Parseval theorem: 'tsum_sq_fourierCoeff'
-    Fourier transforms on $\mathrm{L}^1(\R^d)$ and $\mathrm{L}^2(\R^d)$: ''
+    Fourier transform on $\mathrm{L}^1(\R^d)$: 'Real.fourierIntegral'
```
[the remark this answers is not in the document]
Comment (PatrickMassot) [reply docs/undergrad.yaml:535]: Splitting an entry is definitely fine.

File: docs/undergrad.yaml [hunk 4]
```diff
@@ -532,8 +532,10 @@ Measures and integral calculus:
...
+    Fourier transform on $\mathrm{L}^2(\R^d)$: ''
     Plancherel’s theorem: ''
+    Fourier inversion formula: 'Continuous.fourier_inversion'
```
Comment (PatrickMassot) [inline docs/undergrad.yaml:538]: And this is also fine.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**